crates/rattler_index/src/main.rs

```diff
@@ -0,0 +1,135 @@
+    /// Defaults to all platforms available in the channel.
+    #[arg(long, global = true)]
+    target_platform: Option<Platform>,
+}
+
+/// The subcommands for the `rattler-index` CLI.
+#[derive(Subcommand)]
+#[allow(clippy::large_enum_variant)]
+enum Commands {
+    /// Index a channel stored on the filesystem.
+    #[command(name = "fs")]
+    FileSystem {
+        /// The path to the channel directory.
+        #[arg()]
+        channel: std::path::PathBuf,
+    },
+
+    /// Index a channel stored in an S3 bucket.
+    S3 {
+        /// The S3 channel URL, e.g. `s3://my-bucket/my-channel`.
+        #[arg(value_parser = parse_s3_url)]
+        channel: Url,
+
+        /// The endpoint URL of the S3 backend
+        #[arg(
+            long,
+            env = "S3_ENDPOINT_URL",
+            default_value = "https://s3.amazonaws.com"
+        )]
+        endpoint_url: Url,
+
+        /// The region of the S3 backend
+        #[arg(long, env = "S3_REGION", default_value = "eu-central-1")]
+        region: String,
+
+        /// Whether to use path-style S3 URLs
+        #[arg(long, env = "S3_FORCE_PATH_STYLE", default_value = "false")]
+        force_path_style: bool,
+
+        /// The access key ID for the S3 bucket.
+        #[arg(long, env = "S3_ACCESS_KEY_ID", requires_all = ["secret_access_key"])]
+        access_key_id: Option<String>,
+
+        /// The secret access key for the S3 bucket.
+        #[arg(long, env = "S3_SECRET_ACCESS_KEY", requires_all = ["access_key_id"])]
+        secret_access_key: Option<String>,
+
+        /// The session token for the S3 bucket.
+        #[arg(long, env = "S3_SESSION_TOKEN", requires_all = ["access_key_id", "secret_access_key"])]
+        session_token: Option<String>,
+    },
+}
+
+/// Entry point of the `rattler-index` cli.
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    // Parse the command line arguments
+    let cli = Cli::parse();
+
+    tracing_subscriber::FmtSubscriber::builder()
+        .with_max_level(cli.verbose.log_level_filter().as_trace())
+        .init();
+
+    let result = match cli.command {
+        Commands::FileSystem { channel } => {
+            index_fs(channel, cli.target_platform, cli.force, cli.max_parallel).await?;
+            Ok(())
+        }
+        Commands::S3 {
+            channel,
+            region,
+            endpoint_url,
+            force_path_style,
+            access_key_id,
+            secret_access_key,
+            session_token,
+        } => {
+            index_s3(
+                channel,
+                region,
+                endpoint_url,
+                force_path_style,
+                access_key_id,
+                secret_access_key,
+                session_token,
+                cli.target_platform,
+                cli.force,
+                cli.max_parallel,
+            )
+            .await?;
+            Ok(())
+        }
+    };
+    println!("Finished indexing channel...");
```

Comment:
```suggestion
    println!("Finished indexing channel.");
```
`...` usually indicates that something is still coming 😄